src/gui/sdl_mapper.cpp

```diff
@@ -197,19 +197,16 @@ class CContinuousEvent : public CEvent {
 			if (!GetActivityCount()) Active(true);
 		}
 	}
-	void DeActivateEvent(bool ev_trigger) override {
-		if (ev_trigger) {
-			if (activity>0) activity--;
-			if (activity==0) {
-				/* test if still some trigger-activity is present,
-				   adjust the state in this case accordingly */
-				if (GetActivityCount()) RepostActivity();
-				else Active(false);
-			}
-		} else {
-			if (!GetActivityCount()) Active(false);
+	void DeActivateEvent(const bool ev_trigger) override
+	{
+		if (ev_trigger || GetActivityCount() == 0) {
```

Comment:
Not familiar with this whole code, but I'm wondering why just `if (ev_trigger)` is not enough?

Reply:
Yup; that second path exists in the original code, so I've retained it.

If the event is deactived indirectly (when ev_trigger is false), then this activity-count-zero code path will still deactivate the 'held' event if the opposite axis also has no events remaining (the get activity count function includes the opposite direction too). 

In this second code path, if the opposite axis has events remaining then what happens is the joystick can move to the opposite side in the next step (imagine pulling right to left in one go). 

Where as if there are no activities left, then we reactive back to zero (or center). 

This appears to work that way too: I can snap right and left quickly, but also settle it back to center

Reply:
Ok, that's some weird ass logic 😅 I guess this will do until we rebuild the mapper from scratch if it works well enough in the interim 🤷🏻